src/shims/unix/linux/epoll.rs

```diff
@@ -11,7 +13,16 @@ use self::shims::unix::fd::FileDescriptor;
 #[derive(Clone, Debug, Default)]
 struct Epoll {
     /// The file descriptors we are watching, and what we are watching for.
-    file_descriptors: FxHashMap<i32, EpollEvent>,
+    interest_list: BTreeMap<(WeakFileDescriptor, i32), EpollEvent>,
+    ready_list: Rc<RefCell<BTreeMap<(WeakFileDescriptor, i32), EpollReturn>>>,
```

Comment:
Document that this is an `Rc`, because the `EpollEvent`s need to be able to add new `EpollReturn`s (or update them).